.github/workflows/ci.yml

```diff
@@ -163,7 +163,8 @@ jobs:
         run: |
```

Comment:
Hrm I guess we didn't catch the errors because of the piping?  Not sure why `pipefail` doesn't fix that.  I wonder though, should these just be two test targets, "Tests root" and "Tests nonroot"?  Then we don't need any piping to change the output, and less shell magic to wait for processes etc.